protowhat/checks/check_files.py

```diff
@@ -28,13 +28,17 @@ def check_file(
             state.parse(solution_code, test=False) if parse else False
         )
 
-    return state.to_child(
+    child_state = state.to_child(
         append_message="We checked the file `{}`. ".format(path),
         student_code=code,
         student_ast=state.parse(code) if parse else False,
         **sol_kwargs
     )
 
+    child_state.path = path_obj   # .parent + .name
```

Comment:
Hmm... why not allow `path` in the state constructor instead of this?

Reply:
Ideally, the Liskov substitution principle applies to the State class across all xwhats.
Adding path to State in protowhat would mean adding it to all xwhats, while the meaning across all xwhats wasn't immediately clear.

The downside is that you can't assume that it will exist, so if you want to use it, more safety checks are needed (existence + usable value).

One place that State.path could be useful is the highlighting. But that ultimately means that downstream it will be there across all technologies (but as a null value depending on the exercise and technology).

Reply:
Thinking more about it, it could be a database name for sql, a sheet name... (if we can get that information).

Reply:
But maybe there's a better name for the attribute then. (U)RI, name...